gantry/views.py

```diff
@@ -0,0 +1,25 @@
+import json
+import os
+
+from aiohttp import web
+
+from gantry.collection import fetch_build
+
+routes = web.RouteTableDef()
+
+
+@routes.post("/collect")
+async def collect_job(request: web.Request) -> web.Response:
+    try:
+        payload = await request.json()
+    except json.decoder.JSONDecodeError:
+        return web.Response(status=400, text="invalid json")
+
+    if request.headers.get("X-Gitlab-Token") != os.environ["GITLAB_WEBHOOK_TOKEN"]:
+        return web.Response(status=401, text="invalid token")
+
+    if request.headers.get("X-Gitlab-Event") != "Job Hook":
+        return web.Response(status=400, text="invalid event type")
```

Comment:
I had this in mind when writing that

> Only return client error status responses (in the 4xx range) to indicate that the webhook has been misconfigured. Responses in this range can lead to your webhooks being [automatically disabled](https://docs.gitlab.com/ee/user/project/integrations/webhooks.html#failing-webhooks). For example, if your receiver only supports push events, you can return 400 if sent an issue payload, as that is an indication that the hook has been set up incorrectly. Alternatively, you can ignore unrecognized event payloads.

From the [docs](https://docs.gitlab.com/ee/user/project/integrations/webhooks.html#configure-your-webhook-receiver-endpoint)

I'd be happy to log and ignore the unrecognized event type, but I was thinking it might be better to treat auth issues as failures